src/indentMain.ml

```diff
@@ -69,16 +69,22 @@ let indent_file args = function
           config
           args.Args.indent_config
       in
-      let out, need_move =
+      let out, perm, need_move =
         if args.Args.inplace then
           let tmp_file = path ^ ".ocp-indent-tmp" in
-          Some tmp_file, Some path
+          let rec get_true_file path =
+            let open Unix in
+            match lstat path with
+            | { st_kind = S_REG ; st_perm } -> Some tmp_file, st_perm, Some path
+            | { st_kind = S_LNK ; } -> get_true_file @@ readlink path
+            | { st_kind = _ ; } -> failwith "invalid file type"
+          in get_true_file path
         else
-          args.Args.file_out, None
+          args.Args.file_out, 0o644, None
       in
       let ic = open_in path in
       try
-        indent_channel ic args config out;
+        indent_channel ic args config out perm;
```

Comment:
Not relevant to this PR, but it seems that the channel `ic` is not closed in the usual (non-exceptional) case.

Reply:
Thank you for pointing this out! I'm on it.